source/GameInterface/DynamicSync/Builders/DynamicSyncFieldBuilder.cs

```diff
@@ -0,0 +1,118 @@
+                {
+                    MemberDeclaringType = fieldInfo.DeclaringType.Name,
+                    MemberName = fieldInfo.Name,
+                    MemberType = fieldInfo.FieldType.Name,
+                    Libraries = new List<string>
+                    {
+                        fieldInfo.DeclaringType.Namespace,
+                        fieldInfo.FieldType.Namespace
+                    }
+                });
+            string networkMessage;
+            if (objectManager.IsTypeManaged(fieldInfo.FieldType))
+            {
+                networkMessage = TemplateParser.Parse("Messages.NetworkSetReferenceMessageTemplate",
+                    new
+                    {
+                        MemberDeclaringType = fieldInfo.DeclaringType.Name,
+                        MemberName = fieldInfo.Name,
+                        MemberType = fieldInfo.FieldType.Name,
+                        Libraries = new List<string>
+                        {
+                        fieldInfo.DeclaringType.Namespace,
+                        fieldInfo.FieldType.Namespace
+                        }
+                    });
+            }
+            else
+            {
+                networkMessage = TemplateParser.Parse("Messages.NetworkSetValueMessageTemplate",
+                    new
+                    {
+                        MemberDeclaringType = fieldInfo.DeclaringType.Name,
+                        MemberName = fieldInfo.Name,
+                        MemberType = fieldInfo.FieldType.Name,
+                        Libraries = new List<string>
+                        {
+                            fieldInfo.DeclaringType.Namespace,
+                            fieldInfo.FieldType.Namespace
+                        }
+                    });
+            }
+
+            DynamicSyncConfiguration.ExportFile($"{fieldInfo.DeclaringType.Name}/{fieldInfo.DeclaringType.Name}_{fieldInfo.Name}_SetLocalMessage.cs", localMessage);
+            DynamicSyncConfiguration.ExportFile($"{fieldInfo.DeclaringType.Name}/{fieldInfo.DeclaringType.Name}_{fieldInfo.Name}_SetNetworkMessage.cs", networkMessage);
+
+            yield return localMessage;
+            yield return networkMessage;
+        }
+
+        public string GetSubscription(FieldInfo fieldInfo)
+        {
+            if (objectManager.IsTypeManaged(fieldInfo.FieldType))
+            {
+                return TemplateParser.Parse("Handlers.SubscribeGenericSetReferenceTemplate",
+                    new
+                    {
+                        MemberDeclaringType = fieldInfo.DeclaringType.Name,
+                        MemberName = fieldInfo.Name,
+                        MemberType = fieldInfo.FieldType.Name,
+                        Libraries = new List<string>
+                        {
+                            fieldInfo.DeclaringType.Namespace,
+                            fieldInfo.FieldType.Namespace
+                        }
+                    });
```

Comment:
Store the template name in a variable and you can parse after the if statement so you don't need so many lines